modules/gapi/samples/pipeline_modeling_tool/dummy_source.hpp

```diff
@@ -46,7 +49,9 @@ bool DummySource::pull(cv::gapi::wip::Data& data) {
     auto end = high_resolution_clock::now();
     auto elapsed =
         duration_cast<duration<double, std::milli>>(end - *m_prev_pull_tp).count();
-    auto delta = m_latency - elapsed;
+
+    auto delta =
+        (m_drop_frames ? (static_cast<int>(elapsed / m_latency) + 1) * m_latency : m_latency) - elapsed;
     if (delta > 0) {
         utils::sleep(delta);
     }
```

Comment:
I don't understand what this logic is doing and why.

Anyway, if you drop a frame, I'd expect a `seq_no` be skipped in the sequence - so  a framedrop could later be identified at the app level. Are you sure you tag your frames properly?

Reply:
@mpashchenkov would you fix this?

Reply:
Probably "dropping frames" is confusing...
In fact, this option allows the source to return frames ** only ** at discrete points in time, which are integers divided by 'm_latency`

Moreover, this should work only with `regular` mode, so `seq_no` isn't relevant.

Reply:
I'd not use `?` operator to make a logic clearer:
```
time_t current = now();
time_t expected = 0;
if (drop) {
    // Get the closest divisible by latency
    expected = (current/latency + 1)*latency;
} else {
    expected = prev + latency;
    prev = now();
}
sleep(expected - current);
```
And here is incorrect behavior possible if current time is perfectly divisible by latency, we'll get +latency value in this case, but the probability is very low. (Need to change to (current-1)/latency to eliminate this case)

And this stream source definitely misses some comments on what this frame drop means

Reply:
Discussed locally